src/ol/expr/gpu.js

```diff
@@ -56,11 +56,19 @@ export function arrayToGlsl(array) {
  * Will normalize and converts to string a `vec4` color array compatible with GLSL.
  * @param {string|import("../color.js").Color} color Color either in string format or [r, g, b, a] array format,
  * with RGB components in the 0..255 range and the alpha component in the 0..1 range.
- * Note that the final array will always have 4 components.
+ * Alternatively only 2 element array can be supplied as [shade, alpha] for grayscale colors.
+ * Alpha value is optional. Note that the final array will always have 4 components.
  * @return {string} The color expressed in the `vec4(1.0, 1.0, 1.0, 1.0)` form.
  */
 export function colorToGlsl(color) {
   const array = asArray(color);
+  if (array.length < 3) {
+    if (array.length == 2) {
+      array[3] = array[1];
+    }
+    array[1] = array[0];
+    array[2] = array[0];
+  }
```

Comment:
It is just for consistency, I was not sure whether it can somehow receive the one/two element array through the `compile` function in gpu.js where it is used.

Reply:
As far as I'm aware this should only be used for simple colors with numerical values. So I would remove this change from the PR as it might be misleading.

Reply:
OK